enterprise/server/backends/gcs_cache/gcs_cache.go

```diff
@@ -473,10 +474,15 @@ func (wc *gcsDedupingWriteCloser) Write(in []byte) (int, error) {
 	return n, err
 }
 
-func (wc *gcsDedupingWriteCloser) Close() error {
+func (wc *gcsDedupingWriteCloser) Commit() error {
 	return swallowGCSAlreadyExistsError(wc.WriteCloser.Close())
```

Comment:
wc.WriteCloser.Commit? And wc.WriteCloser.Close should be added to line 481?

Reply:
The underlying io.WriteCloser comes from the GCS storage library so it doesn't have a commit() method -- it finalizes a write when Close() is called.

To work around this, we call Close() on the gcs WriteCloser when Commit() is called, and when Close() is called, we cancel the context given to the gcs WriteCloser.

This means that if Close is called but Commit is not, the file will *not* be written, but resources will be cleaned up. If both are called, the file will be written and resources will be cleaned up.